.github/workflows/lint.yaml

```diff
@@ -24,8 +24,13 @@ jobs:
         with:
           python-version: 3.7
 
-      - name: Pre-Commit Checks
```

Comment:
Why not both?

Otherwise we loose:
-   id: trailing-whitespace
-   id: end-of-file-fixer
-   id: check-yaml

Reply:
Also suggest not to drive by CI changes

Reply:
The problem is that we cannot run a straight `pre-commit` check, because we need to weaken some rules.
I have basically edited this using what we are already doing for `tools`

Reply:
@irq0 are we fine keeping these changes?

Reply:
Suggest to weaken the individual rules isteand of loosing the yaml check and the basic whitespace checks in a yaml heavy repo. 

Also suggest to not mix in CI changes with code changes in the same commit.  For instance, it makes cherry picking needlessly complicated.

Reply:
re-added pre-commit checks.
As expected CI is now failing with:
```
CHANGELOG.md:55 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "### Added"]
CHANGELOG.md:74 MD024/no-duplicate-heading/no-duplicate-header Multiple headings with the same content [Context: "### Removed"]
```
suggestion how to fix this?

Reply:
@irq0 that was super-easy :smirk: : 
```
git mv .github/workflows/config/custom.markdownlint.jsonc .markdownlint.jsonc
```
credits to @m-ildefons :top:

I've removed the ugly Lint reading the config file, now the exception is automatically handled by the pre-commit check.

Reply:
:+1: